2.train_model/scripts/nbconverted/train_multi_class_models.py

```diff
@@ -55,64 +61,89 @@
 
 # ### Train model on each combination of model type and feature type
 
-# In[3]:
+# In[4]:
 
 
 # specify model types and feature types
 model_types = ["final", "shuffled_baseline"]
-feature_types = ["CP", "DP", "CP_and_DP"]
-
-# create stratified data sets for k-fold cross validation
-straified_k_folds = StratifiedKFold(n_splits=10, shuffle=False)
-
-# create logistic regression model with following parameters
-log_reg_model = LogisticRegression(
-    penalty="elasticnet", solver="saga", max_iter=100, n_jobs=-1, random_state=0
-)
+feature_types = ["CP", "DP", "CP_and_DP", "CP_zernike_only", "CP_areashape_only"]
+balanced_types = ["balanced", "unbalanced"]
 
 # specify parameters to tune for
 parameters = {"C": np.logspace(-3, 3, 7), "l1_ratio": np.linspace(0, 1, 11)}
 print(f"Parameters being tested during grid search: {parameters}\n")
 
-# create grid search with cross validation with hypertuning params
-grid_search_cv = GridSearchCV(
-    log_reg_model,
-    parameters,
-    cv=straified_k_folds,
-    n_jobs=-1,
-    scoring="f1_weighted",
-)
-
-# train model on each combination of model type and feature type
-for model_type in model_types:
-    for feature_type in feature_types:
-        print(f"Training {model_type} model on {feature_type} features...")
-
-        X, y = get_X_y_data(training_data, feature_type)
-        print(f"X has shape {X.shape}, y has shape {y.shape}")
-
-        # shuffle columns of X (features) dataframe independently to create shuffled baseline
-        if model_type == "shuffled_baseline":
-            for column in X.T:
-                np.random.shuffle(column)
-
-        # fit grid search cv to X and y data
-        # capture convergence warning from sklearn
-        # this warning does not affect the model but takes up lots of space in the output
-        with parallel_backend("multiprocessing"):
-            with warnings.catch_warnings():
-                warnings.filterwarnings(
-                    "ignore", category=ConvergenceWarning, module="sklearn"
+# train model on each combination of model type, feature type, and balance type
+for balance in balanced_types:
+    if balance == "balanced":
```

Comment:
I suggest adding a short code comment on why you are specifying the "balanced" model and setting the "unbalanced" model to `None`. I am assuming so that it doesn't run this model but I am not 100% sure.

Reply:
Sounds good! A short comment added in 7f2ca61